### ISSUE TITLE
Run sass in parallel

### ISSUE THREAD

**Issue by chrisirhc** [gruntjs/grunt-contrib-sass#16]
I'm wondering if there are any repercussions for running `sass` in parallel. This is a whole performance bump when running the sass task for many files.

I see in https://github.com/gruntjs/grunt-contrib-sass/blob/aa0f821697e3f1a5046ae4ed50e9358967d9bdd7/tasks/sass.js#L21:

```
    grunt.util.async.forEachSeries(this.files, function(f, next) {
```

I can understand there will be issues if some of the sass tasks are writing to files that are input files for other sass tasks but that seem very unlikely. Some questions that might help in this decision:
1. Are there a cases where a user runs sass on a .css file?
2. Are there cases where _within the same task_ as user writes _and_ reads from the same file?

The answer to the first question is could be _maybe_ since running sass on css can still return a valid css file.

As for question 2, I'm hoping that that this type of task definition/configuration would be highly discouraged (or even considered invalid). Why? Because it relies on the ordering of the keys in order to be deterministic. node.js respects insertion order for keys, but [this shouldn't be relied upon](http://stackoverflow.com/questions/9179680/is-it-acceptable-style-for-node-js-libraries-to-rely-on-object-key-order).

This is an easy one liner fix to support an option for running it in parallel with a limit of 10. An `async` or `parallel` option can used.

```
    grunt.util.async.forEachLimit(this.files, 10, function(f, next) {
```

node warns of a memory leak if it is set to anything higher than 10, but that's a separate issue.

In my project that requires 55 destination css files (it's a big project), this decreased the task's running time from 14 seconds to 3 seconds.
